packages/gguf/src/gguf.ts

```diff
@@ -185,8 +188,8 @@ export interface GGUFParseOutput {
 	tensorInfos: GGUFTensorInfo[];
 }
 
-export async function gguf(url: string): Promise<GGUFParseOutput> {
-	const r = new RangeView(url);
+export async function gguf(url: string, _fetch: typeof fetch = fetch): Promise<GGUFParseOutput> {
```

Comment:
```suggestion
export async function gguf(url: string, params: {
		/**
		 * Custom fetch function to use instead of the default one, for example to use a proxy or edit headers.
		 */
		fetch?: typeof fetch;
}): Promise<GGUFParseOutput> {
```

any way to pass it like this for consistency with safetensors?

Reply:
https://github.com/huggingface/huggingface.js/pull/543/commits/682483a86e899d65a5dacd186e0a89bcdb91867c